src/gather.ts

```diff
@@ -150,37 +151,50 @@ export class GatherProvider implements IGatherProvider {
 
         // Call internal slice method
         const slice = this._executionSlicer.sliceLatestExecution(gatherCell.persistentId);
-        const PpaResult = slice.cellSlices
-          .reduce(concat, "")
-          .replace(/#%%/g, defaultCellMarker)
-          .trim();
 
-        if (PpaResult.length === 0) {
-          vscode.window.showErrorMessage(localize.Common.gatherError()  + '\n' + localize.Common.PPAError());
-          return "";
-        }
-
-        return PpaResult;
-      // }
+        if (slice) {
+          const PpaResult = slice.cellSlices
+            .reduce(concat, "")
+            .replace(/#%%/g, defaultCellMarker)
+            .trim();
 
-      // if (vscCell.language === 'C#') {
-      //   C# work
-      // }
+          if (PpaResult.length === 0) {
+            vscode.window.showErrorMessage(localize.Common.gatherError()  + '\n' + localize.Common.PPAError());
+            return "";
+          }
 
-      // return defaultCellMarker + localize.Common.notAvailable() + ' in ' + vscCell.language;
+          return PpaResult;
+        }
+        
+        sendTelemetryEvent(Telemetry.GatherException, undefined, { exceptionType: 'gather' });
+        return defaultCellMarker + newline + localize.Common.gatherError() + newline + localize.Common.couldNotAnalyze();
     } catch (e) {
-      // Cannot read property 'cellSlices' of undefined
-      if ((e.message as string).includes('cellSlices') && e.message.includes('undefined')) {
-        vscode.window.showInformationMessage(localize.Common.runCells());
-        return "";
-      }
-      vscode.window.showErrorMessage(localize.Common.gatherError());
       console.error(e);
       sendTelemetryEvent(Telemetry.GatherException, undefined, { exceptionType: 'gather' });
       return defaultCellMarker + newline + localize.Common.gatherError() + newline + (e as string);
     }
   }
 
+  public async gatherWithoutKernel(vscCell: vscode.NotebookCell, toScript: boolean): Promise<void> {
+    await this.initPromise;
+
+    // Log all the cells up to vcsCell
+    for (let index = 0; index < vscCell.notebook.getCells().length; index++) {
+      const cell = vscCell.notebook.cellAt(index);
+      await this.logExecution(cell);
+
+      if (cell.index === vscCell.index) {
+        break;
+      }
+    }
+
+    // Gather normally
+    await this.gatherCode(vscCell, toScript);
+
+    // End by reseting the log
+    await this.resetLog();
```

Comment:
Should the reset be in a try/finally block?

Reply:
good idea